src/handshake/client.rs

```diff
@@ -135,10 +136,12 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Client<'a, T> {
 		self.buffer.extend_from_slice(b"\r\nUpgrade: websocket\r\nConnection: Upgrade");
 		self.buffer.extend_from_slice(b"\r\nSec-WebSocket-Key: ");
 		self.buffer.extend_from_slice(&self.nonce);
-		if let Some(o) = &self.origin {
-			self.buffer.extend_from_slice(b"\r\nOrigin: ");
-			self.buffer.extend_from_slice(o.as_bytes())
-		}
+		self.headers.iter().for_each(|h| {
+			self.buffer.extend_from_slice(b"\r\n");
+			self.buffer.extend_from_slice(h.name.as_bytes());
+			self.buffer.extend_from_slice(b": ");
+			self.buffer.extend_from_slice(h.value);
```

Comment:
I'm just wondering here; should we do anything about custom header values possibly containing newlines and such, or perhaps add a comment somewhere to note that we shouldn't accept untrusted input or something?

Reply:
Ah, now I read it more, I can see that they probably are sanitzed via httparse!

Reply:
I'm fine with adding a note since it takes a slice of a type with public fields.
I.e. I had `jsonrpsee` with 
```	/// Append an custom header to the header list.
	pub fn custom_header(mut self, name: &'a str, value: &'a [u8]) -> Self {
		self.headers.push(Header{name, value});
		self
	}
```
which essentially doesn't perform any checks. I'm a big fan of `http`'s `HeaderMap`, both for performance and for its' clean API. Requires a bit more space for the construction of the map, however.